server/router.go

```diff
@@ -144,21 +150,39 @@ func (router *PubSubRouter) unsubscribe(r *Route) {
 	}
 }
 
-func (router *PubSubRouter) HandleMessage(message *guble.Message) error {
+func (router *router) HandleMessage(message *guble.Message) error {
 	guble.Debug("Route.HandleMessage: %v %v", message.PublisherUserId, message.Path)
 	if !router.accessManager.IsAllowed(auth.WRITE, message.PublisherUserId, message.Path) {
 		return &PermissionDeniedError{message.PublisherUserId, auth.WRITE, message.Path}
 	}
 
+	return router.storeTxHandle(message)
+}
+
+// Assign the new message id and store it and handle by passing the stored message
+// to the messageIn channel
+func (router *router) storeTxHandle(msg *guble.Message) error {
```

Comment:
I don't like this method name. It does not give a clear understanding of what's going on.
Maybe it's the best to just inline it into HandleMessage. Or just do the storing part here and name it storeMessage